### PR TITLE
Voxel grid model line iterator

### PR DESCRIPTION
**Pull request by efernandez**
This PR makes the `voxel_grid_model` use the `LineIterator` as in the `costmap_model`.
It's just a cosmetic change.

### CODE REVIEW

**Comment by DLu** [comment]
LGTM. Tested I assume?

**Comment by efernandez** [comment]
@DLu Sorry, but I haven't tested.
I'd like that you tell me of a good example to exercise this code. If you can give me an example config file, I'd try it with the REEM in simulation (http://wiki.ros.org/Robots/REEM); I'd take care of adapting it.
I'm using `costmap` type for several observation sources (in different layers/plugins): laser, sonars, IRs, PCs.
Would it be enough if I change to `voxel` the `map_type`?

File: base_local_planner/src/voxel_grid_model.cpp [hunk 1]
```diff
@@ -91,62 +92,13 @@ namespace base_local_planner {
...
-      den = deltay;
-      num = deltay / 2;
-      numadd = deltax;
-      numpixels = deltay;         // There are more y-values than x-values
-    }
-
-    for (int curpixel = 0; curpixel <= numpixels; curpixel++)
+    for( LineIterator line( x0, y0, x1, y1 ); line.isValid(); line.advance() )
     {
       point_cost = pointCost(x, y); //Score the current point
```
Comment (DLu) [inline base_local_planner/src/voxel_grid_model.cpp]: should be line.getX(), line.getY()

**Comment by DLu** [comment]
Careful visual inspection looks good, but did your version compile?

**Comment by efernandez** [comment]
Sure, it compiles, but I haven't tried it with a configuration that uses `voxel`.
If you want, give me some time and I'll try something in a few weeks.

**Comment by mikeferguson** [comment]
Do any of the unit tests cover this function? If so, that should be a good start to run, if not, maybe it would be a good time to add them....

**Comment by DLu** [comment]
The code does not compile as is. It will work if you change the line I suggested (x and y aren't declared in that scope anymore). None of the unit tests specifically test the cost models, and there is nothing that tests the voxel model.

**Comment by efernandez** [comment]
@DLu I've fixed the compilation problem.
Still I need some time to run/test it.
Writing a test will take more time... is there an example I can start with?

**Comment by mikeferguson** [comment]
Hydro is EOL, so we can no longer merge this -- please consider a PR against a newer branch.

**Pull request closed without merging**